File: php/src/JSONKeyNormalizer/FullLoadNormalizer.php
<?php

namespace App\JSONKeyNormalizer;

class FullLoadNormalizer
{
    use NormalizerTrait;

    private array $result = [];
    private array $input  = [];

    public function run(): void
    {
        $this->input = json_decode(file_get_contents(self::INPUT_FILE), true);

        $this->recursiveNormalize($this->input);

        file_put_contents(__DIR__ . "/../../json-normalized.json", json_encode($this->result, JSON_PRETTY_PRINT));
    }

    private function recursiveNormalize(array $data, string $parent_key = ""): void
    {
        $parent_key = $this->prepareParentKey($parent_key);

        foreach ($data as $key => $value) {
            $current_key = $parent_key . $key;
            if (is_array($value)) {
                $this->recursiveNormalize($value, $current_key);
            } else {
                $this->result[$this->normalize($current_key)] = $value;
            }
        }
    }
}

File: c#/Benchmark/Logic/JSONKeyNormalizer/FullLoadNormalizer.cs
using System.Text.Json;

namespace Benchmark.Logic.JSONKeyNormalizer;

public class FullLoadNormalizer : AbstractNormalizer
{
    private readonly Dictionary<string, string>      _result = new();
    private          Dictionary<string, JsonElement> _input  = new();

    public void Run()
    {
        string data = File.ReadAllText(this.InputPath) ?? throw new InvalidOperationException();
        this._input = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(data) ?? throw new InvalidOperationException();

        this.RecursiveNormalize(this._input);
        JsonSerializerOptions options = new() { WriteIndented = true };
        File.WriteAllText(this.OutputPath, JsonSerializer.Serialize(this._result, options));
    }

    private void RecursiveNormalize(Dictionary<string, JsonElement> data, string parentKey = "")
    {
        parentKey = this.PrepareParentKey(parentKey);

        foreach ((string key, JsonElement value) in data) {
            string currentKey = $"{parentKey}{key}";

            if (value.ValueKind == JsonValueKind.Object) {
                this.RecursiveNormalize(value.Deserialize<Dictionary<string, JsonElement>>()!, currentKey);
            } else {
                this._result.TryAdd(this.Normalize(currentKey), value.GetString()!);
            }
        }
    }
}
